pkg/swarm/swarm.go

```diff
@@ -185,3 +185,24 @@ func (c *chunk) WithType(t Type) Chunk {
 type ChunkValidator interface {
 	Validate(ch Chunk) (valid bool)
 }
+
+type chunkValidators struct {
+	set []ChunkValidator
+	ChunkValidator
+}
```

Comment:
why do we need both the `set` field plus the embedded `ChunkValidator` interface?

Reply:
because chunkValidators implements this interface. we might not need to include it explicitly in the struct definition. this was added just for sake of clarity. `set` has the actual validators